roles/setup_lvm_vg/action_plugins/validate_storage.py

```diff
@@ -0,0 +1,470 @@
+                        "size": "10.00 GB",
+                        "support_discard": "0",
+                        "vendor": "0x1af4",
+                        "virtual": 1
+                    }
+                }
+            }
+        }
+    }}
+'''
+
+# Note: to add mode checks/validations, simply add a top-level function whose
+# name starts with `check_`. The function will receive a `task_vars` dictionary
+# as defined by Ansible.
+# Within a `check_*` function, use `assert` to validate values found in
+# `task_vars`, or raise `AssertionError` explicitly. A single check can, of
+# course, contain multiple assertions.
+# Alternatively, for checks which can result in multiple errors, a check can
+# return a list of (or yield) error messages.
+
+from ansible.plugins.action import ActionBase
+
+import inspect
+import sys
+
+
+def is_drive_raw_device(device, ansible_devices):
+    '''Check whether a drive is a raw device or partition
+
+    :param str device: The name of the device
+        i.e: 'sdb' or 'sdb1'
+    :param dict ansible_devices: The dictionary of devices gather by the
+        'setup' ansible module
+    :returns: A tuple with True, and the device details from ansible_devices
+        if the device is a raw device. False otherwise
+    :rtype: tuple
+    '''
+
+    if device in ansible_devices:
+        return True, ansible_devices[device]
+
+    return False
+
+
+def is_drive_partition(device, ansible_devices, unused=False):
+    '''Check whether a drive is a raw device or partition
+
+    :param str device: The name of the device
+        i.e: 'sdb' or 'sdb1'
+    :param dict ansible_devices: The dictionary of devices gather by the
+        'setup' ansible module
+    :returns: True if the device is a raw device. False otherwise
+    :rtype: bool
+    :returns: A tuple with True, and the partition details from ansible_devices
+        if the device is a partition. False otherwise
+    :rtype: tuple
+    '''
+
+    for ansible_device, device_attr in ansible_devices.items():
+        try:
+            return True, device_attr['partitions'][device]
+        except KeyError:
+            continue  # May also pass if device_attr has no 'partitions' key
+
+    return False, {}
+
+
+def is_device_present(device, ansible_devices):
+    '''
+    Check that the device passed in parameters exists
+
+    The device can be either a raw device or a partition but nothing else.
+
+    :param str device: The name of the device
+        i.e: 'sdb' or 'sdb1'
+    :param dict ansible_devices: The dictionary of devices gather by the
+        'setup' ansible module
+    :returns: True if the device is in ansible_devices. False otherwise
+    :rtype: bool
+    '''
+
+    # if the device is a raw device, check its presence
+    if is_drive_partition(device, ansible_devices) \
+       or is_drive_raw_device(device, ansible_devices):
+        return True
+
+    return False, {}
+
+
+def is_used_by_filesystem(device, uuid):
```

Comment:
Sorry to be annoying about that: some function is_* return boolean, other raise AssertionError. Could we use two different
prefix to distinguish the difference between the API ?